### PR TITLE
feat: support for clerk auth 

### PR DESCRIPTION
**Pull request by yashmehrotra**
Merge https://github.com/flanksource/mission-control-chart/pull/98 after this PR is merged

### CODE REVIEW

File: auth/clerk_client.go [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+		// Extract session token from Authorization header
+		sessionToken := c.Request().Header.Get(echo.HeaderAuthorization)
+		sessionToken = strings.TrimPrefix(sessionToken, "Bearer ")
+
+		var (
+			user   *clerk.User
+			err    error
+			sessID string
+		)
```
Comment (adityathebe) [inline auth/clerk_client.go:53]: nit: this can go away

File: auth/clerk_client.go [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+				logger.Errorf("Error creating user in database from clerk: %v", err)
+				return c.String(http.StatusUnauthorized, "Unauthorized")
+			}
+			// Clear user from cache so that new metadata is used
+			h.userCache.Delete(sessID)
+		}
+
+		token, err := h.getDBToken(sessID, *user.ExternalID)
+		if err != nil {
+			logger.Errorf("Error generating JWT Token: %v", err)
```
Comment (adityathebe) [inline auth/clerk_client.go:73]: probably needs to return as well

File: auth/clerk_client.go [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+
+func (h *ClerkHandler) getUser(sessionToken string) (*clerk.User, string, error) {
+	sessClaims, err := h.client.VerifyToken(sessionToken)
+	if err != nil {
+		return nil, "", err
+	}
+
+	cacheKey := sessClaims.SessionID
+	if user, exists := h.userCache.Get(cacheKey); exists {
+		return user.(*clerk.User), "", nil
```
Comment (adityathebe) [inline auth/clerk_client.go:112]: don't we need to return session id in here?

File: auth/clerk_client.go [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+}
+
+func (h ClerkHandler) Session(next echo.HandlerFunc) echo.HandlerFunc {
+	return func(c echo.Context) error {
+		if canSkipAuth(c) {
+			return next(c)
+		}
+
+		// Extract session token from Authorization header
+		sessionToken := c.Request().Header.Get(echo.HeaderAuthorization)
```
Comment (moshloop) [inline auth/clerk_client.go:46]: I think we might also need to extract it via the `_session` cookie

File: auth/clerk_client.go (hunk 3 above)
Comment (yashmehrotra) [reply auth/clerk_client.go]: Good catch

File: auth/clerk_client.go (hunk 1 above)
Comment (yashmehrotra) [reply auth/clerk_client.go]: 👍🏽

**Pull request merged**